src/Aspire.Hosting/ResourceBuilderExtensions.cs

```diff
@@ -549,4 +555,16 @@ public static IResourceBuilder<T> ExcludeFromManifest<T>(this IResourceBuilder<T
     {
         return builder.WithAnnotation(ManifestPublishingCallbackAnnotation.Ignore);
     }
+
+#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
+#pragma warning disable RS0016 // Add public types and members to the declared API
+    public static IResourceBuilder<T> WithRelationship<T>(
```

Comment:
I think we should consider having an overload of this which takes an Action<RelationshipAnnotation> which allows developers to stuff some extra data into the relationship property bag. There isn't an easy way to get to it otherwise.

Reply:
IMO we could remove properties. Nothing uses it. And we could add it later once we identify a use for it.

Reply:
Thats fair. Keep the API lean until we need it. Agreed!

Reply:
One place where something like this could be used is when we add references to Azure Resources, we could add some information about how the application is going to use the Azure Resource. For example, when adding a reference from a .NET app to an Azure Storage account, we could add that this app will only ever read from the Storage account. So when we deploy the app, we only give it "read" permissions. Similar for other resources as well - i.e. ServiceBus (which app is the reader and which is the writer).

Reply:
I would imagine it would be modelled as information on the "relationship". Can you have an annotation on an annotation?